week4/paramikolab02.py

```diff
@@ -0,0 +1,41 @@
+import time
+import paramiko
+
+USERNAME = "admin"
+PASSWORD = "cisco"
+
+routers = [1, 2, 3]
+devices_ip = "172.31.101.%s"
+
+command = {
+    "r1": ["conf t", "ip access-list extended Block-telnet/SSH", "deny tcp any any eq 22", "deny tcp any any eq 23", \
+        "permit ip any any", "int g0/1", "ip access-group Block-telnet/SSH in", "int g0/2", "ip add 172.31.101.34 255.255.255.240", \
+        "no shut", "ip access-group Block-telnet/SSH in", "ex", "router ospf 1 vrf Control-DataVRF", "router-id 1.1.1.1", \
+        "network 172.31.101.16 0.0.0.15 area 0", "network 172.31.101.32 0.0.0.15 area 0", "network 1.1.1.1 0.0.0.0 area 0", "do wr"],
+    "r2": ["conf t", "ip access-list extended Block-telnet/SSH", "deny tcp any any eq 22", "deny tcp any any eq 23", \
+        "permit ip any any", "int g0/1", "ip add 172.31.101.33 255.255.255.240", "no shut", "ip access-group Block-telnet/SSH in", \
+        "int g0/2", "ip add 172.31.101.50 255.255.255.240", "ip access-group Block-telnet/SSH in", "no shut", "ex", \
+        "router ospf 1 vrf Control-DataVRF", "router-id 2.2.2.2", "network 172.31.101.32 0.0.0.15 area 0", "network 172.31.101.48 0.0.0.15 area 0", \
+        "network 2.2.2.2 0.0.0.0 area 0", "ex", "do wr"],
+    "r3": ["conf t", "ip access-list extended Block-telnet/SSH", "deny tcp any any eq 22", "deny tcp any any eq 23", \
+        "permit ip any any", "int g0/1", "ip add 172.31.101.49 255.255.255.240", "no shut", "ip access-group Block-telnet/SSH in", \
+        "int g0/2", "ip add dhcp", "no shut", "ip access-group Block-telnet/SSH in", "ip route vrf Control-DataVRF 0.0.0.0 0.0.0.0 192.168.122.1", \
+        "router ospf 1 vrf Control-DataVRF", "router-id 3.3.3.3", "default-information originate", "network 172.31.101.48 0.0.0.15 area 0", \
+        "network 3.3.3.3 0.0.0.0 area 0", "ex", "access-list 1 permit any", "ip nat inside source list 1 int g0/2 vrf Control-DataVRF overload", \
+        "int g0/2", "ip nat outside", "int g0/1", "ip nat inside", "do wr"]
+}
```

Comment:
I think you can split this variable to file such as R1.txt R2.txt etc., this variable is so hard to read it. And if you split to each file, you can use foreach with each file and then if you want to change any configuration it will easier than one line configuration.